variable/trigonometry.cpp

```diff
@@ -30,16 +29,17 @@ Variable sin(Variable &&var) {
 
 Variable &sin(const Variable &var, Variable &out) {
   core::expect::unit_any_of(var, {units::rad, units::deg});
-  copy(var, out);
-  if (var.unit() == units::deg)
-    out *= deg_to_rad;
-  transform_in_place(out, out, element::sin_out_arg);
+  if (var.unit() == units::deg) {
+    transform_in_place(out, to_unit(var, units::rad), element::sin_out_arg);
+  } else {
+    transform_in_place(out, var, element::sin_out_arg);
+  }
```

Comment:
This generates twice as much code, write using, e.g., ternary, or (better) implement #1819 (which is trivial) and just use `to_unit` everywhere.